src/pages/academy/gameSimulator/subcomponents/chapterSimulator/ChapterSimulator.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button } from '@blueprintjs/core';
+import { useRequest } from 'src/commons/utils/Hooks';
+import SourceAcademyGame from 'src/features/game/SourceAcademyGame';
+import { gameSimulatorMenuConstants } from 'src/features/gameSimulator/GameSimulatorConstants';
+import GameSimulatorMenu from 'src/features/gameSimulator/GameSimulatorMenu';
+import { fetchTextAssets } from 'src/features/gameSimulator/GameSimulatorService';
+
+import ChapterSimulatorTextLoader from './ChapterSimulatorTextLoader';
+
+/**
+ * This component renders the Chapter Simulator component in the Game Simulator.
+ *
+ * It will simulate a game chapter using the given text files (from either S3 or Local).
+ *
+ * @param textAssets List of all text assets on S3 to choose from.
+ */
+const ChapterSimulator = () => {
```

Comment:
Ditto on typing with `React.FC`

Reply:
Addressed in [962b341](https://github.com/source-academy/frontend/pull/2836/commits/962b3413e413ff74052dd0e7f90ff5830ec709fb).